server/client.go

```diff
@@ -146,6 +147,15 @@ func (c *client) GetOpts() *clientOpts {
 	return &c.opts
 }
 
+func (c *client) GetTLSConnectionState() *tls.ConnectionState {
+	tc, ok := c.nc.(*tls.Conn)
+	if !ok {
+		return nil
+	}
+	state := tc.ConnectionState()
```

Comment:
Understood, but we do not know that it will be read only. I like the idea of wanting to use TLS as an auth mechanism, just think there may be a better way.